File: Bolts/app/src/main/java/com/example/arnassmicius/bolts/MainActivity.java
package com.example.arnassmicius.bolts;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import bolts.Task;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Task<String> successful = Task.forResult("The good result.");

        Task<String> failed = Task.forError(new RuntimeException("An error message."));
    }
}

File: Lottie/app/src/main/java/com/example/arnassmicius/lottieapp/MainActivity.kt
package com.example.arnassmicius.lottieapp

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btLoading.setOnClickListener { setLoadingAnimation() }
        btSuccess.setOnClickListener { setSuccessAnimation() }
        btFail.setOnClickListener { setFailAnimation() }
    }

    private fun setLoadingAnimation() {
        animationView.setMaxFrame(239)
        animationView.loop(true)
        animationView.playAnimation()
    }

    private fun setSuccessAnimation() {
        animationView.setMaxFrame(418)
        animationView.loop(false)
    }

    private fun setFailAnimation() {
        animationView.setMinFrame(500)
    }
}
